atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java

```diff
@@ -48,19 +55,32 @@
     public void sweep(Collection<WriteInfo> writes, Sweeper sweeper) {
         Map<TableReference, Map<Cell, Long>> maxTimestampByCell = writesPerTable(writes, sweeper);
         for (Map.Entry<TableReference, Map<Cell, Long>> entry : maxTimestampByCell.entrySet()) {
-            Iterables.partition(entry.getValue().keySet(), SweepQueueUtils.BATCH_SIZE_KVS)
-                    .forEach(cells -> {
-                        Map<Cell, Long> maxTimestampByCellPartition = cells.stream()
-                                .collect(Collectors.toMap(Function.identity(), entry.getValue()::get));
-                        follower.run(entry.getKey(), maxTimestampByCellPartition.keySet());
-                        if (sweeper.shouldAddSentinels()) {
-                            kvs.addGarbageCollectionSentinelValues(entry.getKey(),
-                                    maxTimestampByCellPartition.keySet());
-                            kvs.deleteAllTimestamps(entry.getKey(), maxTimestampByCellPartition, false);
-                        } else {
-                            kvs.deleteAllTimestamps(entry.getKey(), maxTimestampByCellPartition, true);
-                        }
-                    });
+            try {
+                Iterables.partition(entry.getValue().keySet(), SweepQueueUtils.BATCH_SIZE_KVS)
+                        .forEach(cells -> {
+                            Map<Cell, Long> maxTimestampByCellPartition = cells.stream()
+                                    .collect(Collectors.toMap(Function.identity(), entry.getValue()::get));
+                            follower.run(entry.getKey(), maxTimestampByCellPartition.keySet());
+                            if (sweeper.shouldAddSentinels()) {
+                                kvs.addGarbageCollectionSentinelValues(entry.getKey(),
+                                        maxTimestampByCellPartition.keySet());
+                                kvs.deleteAllTimestamps(entry.getKey(), maxTimestampByCellPartition, false);
+                            } else {
+                                kvs.deleteAllTimestamps(entry.getKey(), maxTimestampByCellPartition, true);
+                            }
+                        });
+            } catch (IllegalArgumentException e) {
+                // hack because InMemoryKVS doesn't throw a TableMappingNotFoundException...
+                if (e.getCause() instanceof TableMappingNotFoundException
```

Comment:
Is there any other reason that a `TableMapingNotFoundryException` could be thrown?  Could a table be renamed and the name in the TS queue is now out of date?

Reply:
We allow for renames?

Reply:
There's some stuff around tablename encoding or case (see for example https://github.com/palantir/atlasdb/pull/3278), but I would expect all of it to be encapsulated in the KVS and for any `TableMappingNotFoundException`s to be legitimate (or at least, I would hope, given the few related things I've seen...). Oracle has some sort of table name mapping stuff as well, but it again is all underneath the KVS layer.